editor/scene/sprite_frames_editor_plugin.cpp

```diff
@@ -1843,9 +1843,7 @@ void SpriteFramesEditor::_fetch_sprite_node() {
 	}
 
 	bool show_node_edit = false;
-	AnimatedSprite2D *as2d = Object::cast_to<AnimatedSprite2D>(selected);
-	AnimatedSprite3D *as3d = Object::cast_to<AnimatedSprite3D>(selected);
-	if (as2d || as3d) {
+	if (selected->has_method("get_sprite_frames")) {
```

Comment:
This is using `->call("get_sprite_frames")` immediately below, so this is less code and more straightforward of a check. Technically could allow user-made AnimatedSprite classes, but that wasn't a goal here, just a happy consequence.